clcache.py

```diff
@@ -637,8 +648,11 @@ def __init__(self, fileName):
 
     def save(self):
         if self._dirty:
-            with open(self._fileName, 'w') as f:
+            tempFileName = self._fileName + ".new"
+            with open(tempFileName, 'w') as f:
                 json.dump(self._dict, f, sort_keys=True, indent=4)
+            atomicRename(tempFileName, self._fileName)
```

Comment:
Seeing this repetition makes me wonder whether it might be nice to have a little context manager like

```python
@contextlib.contextmanager
def atomicallyWritten(fn):
    tempFileName = fn + ".new"
    with open(tempFileName, 'w') as f:
        yield f
    os.replace(tempFileName, fn)
```

...so you could just replace `open(self._fileName, 'w')` with `atomicallyWritten(self._fileName)`. Maybe not worth it, given that it's just two occurrences. I'm mostly thinking out loud.

Reply:
It makes sense, let's unify it for the moment, it can be unfolded later on if it is not needed anymore.